providers/gov/nih/normalizer.py

```diff
@@ -214,15 +280,52 @@ def maybe_org(self, obj):
         return obj
 
     def get_organization_ctx(self, ctx):
+        # TODO lauren: filter NIL
         org_ctx = {
-            'ORG_NAME': ctx['ORG_NAME'],
-            'ORG_FIPS': ctx['ORG_FIPS'],
-            'ORG_DEPT': ctx['ORG_DEPT'],
-            'ORG_DUNS': ctx['ORG_DUNS'],
-            'ORG_DISTRICT': ctx['ORG_DISTRICT'],
-            'ORG_CITY': ctx['ORG_CITY'],
-            'ORG_STATE': ctx['ORG_STATE'],
-            'ORG_ZIPCODE': ctx['ORG_ZIPCODE'],
-            'ORG_COUNTRY': ctx['ORG_COUNTRY']
+            'ORG_NAME': filter_nil(ctx['ORG_NAME']),
+            'ORG_FIPS': filter_nil(ctx['ORG_FIPS']),
+            'ORG_DEPT': filter_nil(ctx['ORG_DEPT']),
+            'ORG_DUNS': filter_nil(ctx['ORG_DUNS']),
+            'ORG_DISTRICT': filter_nil(ctx['ORG_DISTRICT']),
+            'ORG_CITY': filter_nil(ctx['ORG_CITY']),
+            'ORG_STATE': filter_nil(ctx['ORG_STATE']),
+            'ORG_ZIPCODE': filter_nil(ctx['ORG_ZIPCODE']),
+            'ORG_COUNTRY': filter_nil(ctx['ORG_COUNTRY'])
         }
         return org_ctx
+
+    def get_pi(self, ctx, primary=True):
+        '''
+            <PI>
+                <PI_NAME>VIDAL, MARC  (contact)</PI_NAME>
+                <PI_ID>2094159 (contact)</PI_ID>
+            </PI>
+        '''
+        pi_list = ctx['PIS']['PI'] if type(ctx['PIS']['PI']) is list else [ctx['PIS']['PI']]
+        org_ctx = self.get_organization_ctx(ctx)
+        # only one or none so primary is that one
+        if len(pi_list) <= 1 and primary:
+            pi_list[0]['org_ctx'] = org_ctx
+            return pi_list
+        # more than one, get the primary
+        elif len(pi_list) > 1 and primary:
+            primary_pi = []
+            for pi in pi_list:
```

Comment:
If you're looking for a single result `next(...)` might be a bit nicer to read. (Just remember to catch the `StopIterationError`)

It also makes the error/null case handling a bit more explicit. This block looks like it could return a list from `0` to `n` at first glance.